Construct from MPC obs_sbn file

astro-sobrien opened the pull request:
This branch will allow adler to pull in observations from the MPC obs_sbn file (and mpc_orbits) for analysis

## Review Checklist for Source Code Changes

- [ ] Does pip install still work?
- [x] Have you written a unit test for any new functions?
- [x] Do all the units tests run successfully?
- [ ] Does adler run successfully on a test set of input files/databases?
- [x] Have you used black on the files you have updated to confirm python programming style guide enforcement?

jrob93 reviewed (commented):
This all looks pretty good. I just have some minor points to consider before merging:
- AdlerPlanetoid.py, we need to be very clear about the transfer from MPC to Rubin column names, e.g. in populate_observations_from_mpc_obs_sbn: `provid AS ssObjectId, obsid AS diaSourceId` and in populate_MPCORB_from_mpc_obs_sbn: `fullDesignation AS ssObjectId`. There are comments in the code but we could also add a warning to the doc strings and maybe the readme
- objectdata_utilities.py: `mpc_file_preprocessing` operates on the full mpc sbn table? Should we have an option that runs on only a subset of this, e.g. add a query that only grabs a handful of objects? It depends on how we expect users to interact with the mpc bandaid.
- Can we have an option to run the query for `jplhorizons_filename` if required/include the notebook used to make it? So that people can reproduce the results directly from the B612 rubin.sqlite file
- The bandaid file already has the required values so I was able to use `default jplhorizons_filename = None`, should this be the default so that the argument is not explicitly required?

astro-sobrien commented:
I've added warnings to the functions that construct AdlerPlanetoid from the MPC file. This is tracked in #216.

I've improved the efficiency of `mpc_file_preprocessing` as best I can with checks for whether columns already exist. I don't think the checking/updating of the band names can be made any more efficient than it currently is but it currently takes ~3-4 seconds (this is effectively the bulk of this function's runtime on any subsequent run if the file is already in the expected state)

Removed the addition of info from JPL as this is now in the SSSC bandaid, should hopefully be replaced with layup soon, and generally hammering JPL should be discouraged

jrob93 commented:
This all looks good to me, thanks for the updates. I have just made a small change to get_mpc_osb_sbn_test_data.ipynb to remove the jpl_filename argument and neaten up the comments in that cell.

I haven't timed `mpc_file_preprocessing` properly but I'm had ~17s for the first run and then ~1s for subsequent runs so that is really good IMO. I think this is good to merge
